test/widgets/action_sheet_test.dart

```diff
@@ -450,6 +450,39 @@ void main() {
         }
       });
     });
+
+    group('MarkTopicAsReadButton', () {
+      Future<void> setupToTopicActionSheet(WidgetTester tester, {
+        required bool hasUnreadMessages,
+      }) async {
+        addTearDown(testBinding.reset);
+
+        final channel = eg.stream();
+        const topic = 'test topic';
+        await prepare(
+          channel: channel,
+          topic: topic,
+        );
+
+        final message = eg.streamMessage(
+          stream: channel, topic: topic, flags: hasUnreadMessages ? [] : [MessageFlag.read]);
+        if (hasUnreadMessages) {
+          await store.handleEvent(MessageEvent(id: 0, message: message));
+        }
+        await showFromAppBar(tester,
+          channel: channel, topic: topic, messages: [message]);
+      }
+
+      testWidgets('visible if topic has unread messages', (tester) async {
+        await setupToTopicActionSheet(tester, hasUnreadMessages: true);
+        check(find.text('Mark Topic As Read')).findsOne();
+      });
```

Comment:
Let's try to simplify away the helper, we can always add it back if we find a lot of tests reusing the same logic:

```suggestion
      testWidgets('visible if topic has unread messages', (tester) async {
        await prepare();
        final message = eg.streamMessage(stream: someChannel, topic: someTopic);
        await store.handleEvent(MessageEvent(id: 0, message: message));
        await showFromAppBar(tester, messages: [message]);
        check(find.text('Mark Topic As Read')).findsOne();
      });
```

This way we can also skip the flag and the if-statement.